File: A-SOURCE_CODE/B-CLIENT/A-WEB/Administration/src/viewmodels/post-report/post-report.view-model.ts
import {Post} from "../../models/entities/post";
import {Account} from "../../models/entities/account";

/*
* Post report view model which is included in find post report.
* */
export class PostReportViewModel{

    // Index of post report.
    public id: number;

    // Post information
    public post: Post;

    // Owner of post report.
    public owner: Account;

    // Reporter information.
    public reporter: Account;

    // Body of report.
    public body: string;

    // Reason of report.
    public reason: string;

    // When the report was created.
    public created: number;
}

File: A-SOURCE_CODE/A-SERVICE/Administration/Api/ViewModels/ApiPostReport/PostReportViewModel.cs
using Database.Models.Entities;

namespace Administration.ViewModels.ApiPostReport
{
    public class PostReportViewModel
    {
        /// <summary>
        /// Post index.
        /// </summary>
        public Post Post { get; set; }

        /// <summary>
        /// Post report owner.
        /// </summary>
        public Account Owner { get; set; }

        /// <summary>
        /// Who created report.
        /// </summary>
        public Account Reporter { get; set; }

        /// <summary>
        /// Body of report.
        /// </summary>
        public string Body { get; set; }

        /// <summary>
        /// Reason of report.
        /// </summary>
        public string Reason { get; set; }

        /// <summary>
        /// Time when the report is created.
        /// </summary>
        public double Created { get; set; }
    }
}